x-pack/metricbeat/module/containerd/blkio/blkio_integration_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build integration
+// +build integration
+
+package blkio
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/v7/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+)
+
+func TestFetch(t *testing.T) {
+	service := compose.EnsureUp(t, "containerd")
```

Comment:
Wondering if we actually need an integration test since we don't have an actual target endpoint. With this mocked service we actually test against the same response which is more like a unit test. The same check could achieved using `TestData` like at [https://github.com/elastic/beats/blob/e6839abb1bab5a7ff89ebbf6940703ba401696a0/x-pack/metricbeat/module/coredns/stats/stats_test.go#L19](https://github.com/elastic/beats/blob/e6839abb1bab5a7ff89ebbf6940703ba401696a0/x-pack/metricbeat/module/coredns/stats/stats_test.go#L19) which would be faster and more maintainable.